shader/coarse.wgsl

```diff
@@ -152,11 +155,16 @@ fn main(
     // We need to check only prior stages, as if this stage has failed in another workgroup, 
     // we still want to know this workgroup's memory requirement.   
     if local_id.x == 0u {
+        let failed = (atomicLoad(&bump.failed) & (STAGE_BINNING | STAGE_TILE_ALLOC | STAGE_FLATTEN)) != 0u
+            || atomicLoad(&bump.seg_counts) > config.seg_counts_size;
         // Reuse sh_part_count to hold failed flag, shmem is tight
-        sh_part_count[0] = atomicLoad(&bump.failed);
+        sh_part_count[0] = u32(failed);
     }
     let failed = workgroupUniformLoad(&sh_part_count[0]);
-    if (failed & (STAGE_BINNING | STAGE_TILE_ALLOC | STAGE_PATH_COARSE)) != 0u {
+    if failed != 0u {
+        if wg_id.x == 0u && local_id.x == 0u {
+            atomicOr(&bump.failed, STAGE_COARSE);
```

Comment:
What does this represent? I would think this should be `STAGE_{BEFORE_COARSE}`, and then only if the check on `seg_counts` failed

Reply:
I'll make the change, now that I'm thinking these flags will be interpreted by the client, not just to early-out downstream.